pkg/operator2/starter.go

```diff
@@ -37,116 +40,171 @@ const (
+		"openshift-authentication-operator",
+		"", // an informer for non-namespaced resources
+	)
+
+	// short resync period as this drives the check frequency when checking the .well-known endpoint. 20 min is too slow for that.
+	authOperatorConfigInformers := authopinformer.NewSharedInformerFactoryWithOptions(authConfigClient, time.Second*30,
+		authopinformer.WithTweakListOptions(singleNameListOptions("cluster")),
+	)
+
+	operatorClient := &OperatorClient{
+		authOperatorConfigInformers,
+		authConfigClient.OperatorV1(),
+	}
+
+	resourceSyncer := resourcesynccontroller.NewResourceSyncController(
+		operatorClient,
+		kubeInformersForNamespaces,
+		v1helpers.CachedSecretGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		controllerContext.EventRecorder,
+	)
+
+	opCtx.versionRecorder = status.NewVersionGetter()
+	opCtx.kubeClient = kubeClient
+	opCtx.configClient = configClient
+	opCtx.kubeInformersForNamespaces = kubeInformersForNamespaces
+	opCtx.resourceSyncController = resourceSyncer
+	opCtx.operatorClient = operatorClient
+
+	if err := prepareOauthOperator(ctx, controllerContext, opCtx); err != nil {
+		return err
+	}
+	if err := prepareOauthAPIServerOperator(opCtx); err != nil {
+		return err
+	}
+
+	opCtx.informersToRunFunc = append(opCtx.informersToRunFunc, kubeInformersForNamespaces.Start)
+	opCtx.informersToRunFunc = append(opCtx.informersToRunFunc, authOperatorConfigInformers.Start)
+
+	opCtx.controllersToRunFunc = append(opCtx.controllersToRunFunc, resourceSyncer.Run)
+
+	for _, informerToRunFn := range opCtx.informersToRunFunc {
+		informerToRunFn(ctx.Done())
+	}
+
+	for _, controllerRunFn := range opCtx.controllersToRunFunc {
+		go controllerRunFn(ctx, 1)
+	}
+
+	<-ctx.Done()
+	return nil
+}
+
+func prepareOauthOperator(ctx context.Context, controllerContext *controllercmd.ControllerContext, operatorCtx *operatorContext) error {
 	// protobuf can be used with non custom resources
-	oauthClient, err := oauthclient.NewForConfig(controllerContext.ProtoKubeConfig)
+	// protobuf can be used with non custom resources
+	routeClient, err := routeclient.NewForConfig(controllerContext.ProtoKubeConfig)
 	if err != nil {
 		return err
 	}
 
-	configClient, err := configclient.NewForConfig(controllerContext.KubeConfig)
+	// protobuf can be used with non custom resources
+	oauthClient, err := oauthclient.NewForConfig(controllerContext.ProtoKubeConfig)
 	if err != nil {
 		return err
 	}
 
-	kubeInformersNamespaced := informers.NewSharedInformerFactoryWithOptions(kubeClient, resync,
+	kubeInformersNamespaced := informers.NewSharedInformerFactoryWithOptions(operatorCtx.kubeClient, resync,
 		informers.WithNamespace("openshift-authentication"),
 	)
 
-	kubeSystemNamespaceInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, resync,
+	kubeSystemNamespaceInformers := informers.NewSharedInformerFactoryWithOptions(operatorCtx.kubeClient, resync,
```

Comment:
these factories should be shared. Above is another instance.

Reply:
no, this is the first time we use informers for `kube-system` namespace.